### ISSUE TITLE
fprintd & fprintd-pam missing in the base image

### ISSUE THREAD

**Issue by jurf** [fedora-silverblue/issue-tracker#62]
Without these the fingerprint scanner support in GNOME doesn’t work.

**Comment by tpopela** [comment]
Hi @jurf! Looks live it was removed on purpose in https://pagure.io/workstation-ostree-config/c/384502f67d2d895771b8d9eee1847112933cffb2?branch=master with a note that mentioned https://github.com/pbrezina/authselect/issues/48 that cannot be accessed. @cgwalters or @dustymabe do you remember what the problem was? I also see that Dusty mentioned it in the https://pagure.io/workstation-ostree-config/c/7512f9f4af56b8ad4ed4c78d3ac7fb958f2e0b5a.

If you recall the issue, do you know if it was resolved and we should include fprintd-pam in Silverblue?

**Comment by pbrezina** [comment]
About authselect/#48 Authselect does not support nss-altfiles by default which was and probably still is required by Silverblue to work correctly. I have made a scratch build recently that automatically modifies upstream profiles to add altfiles support on Silverblue but it needs some testing. See: https://bugzilla.redhat.com/show_bug.cgi?id=1751417#c5

**Comment by Pobega** [comment]
+1 for this, would be nice to have fingerprint login support out of the box on Silverblue.

**Comment by tpopela** [comment]
I'm curious whether we should enable this on F33+ and see whether it works. @cgwalters @dustymabe any idea (re. https://github.com/fedora-silverblue/issue-tracker/issues/62#issuecomment-633471352)?

**Comment by dustymabe** [comment]
I'll have to lean on @cgwalters or @jlebon here who understand the user handling stuff better than I do.

**Comment by cgwalters** [comment]
A s a general rule I think most "core daemons" shipped in Workstation comps should also be in Silverblue and that would be true here.  We're just blocked on the authselect bug.

**Comment by jlebon** [comment]
> Hi @jurf! Looks live it was removed on purpose in pagure.io/workstation-ostree-config/c/384502f67d2d895771b8d9eee1847112933cffb2?branch=master with a note that mentioned pbrezina/authselect/issues/48 that cannot be accessed.

Looks like the repo was moved. You can see the original issue here: https://github.com/authselect/authselect/issues/48.

An alternative for now is to drop the dependency for authselect in fprintd so at least users can manually configure it. (Looking at the spec, it looks like it just needs it for `%postun` which isn't relevant on Silverblue: https://src.fedoraproject.org/rpms/fprintd/blob/master/f/fprintd.spec#_69).

**Comment by pbrezina** [comment]
I had a patch for Fedora spec file that adds altfiles only on Silverblue but knowbody wanted to test it. Would you be willing to test it? If it works correctly then we can finally resolve the issue.

**Comment by jurf** [comment]
I have no idea what are altfiles, so I can test it, albeit completely blindly. Is it safe to do it that way?

**Comment by pbrezina** [comment]
See https://bugzilla.redhat.com/show_bug.cgi?id=1751417#c9 for scratch build. I'm not very familiar with Silverblue but AFAIK it puts some users in different file then /etc/passwd. nss-altfiles adds support to read this other file and provide users from this file. Without it, the system users are not present and the system is not functioning correctly.

**Comment by YaLTeR** [comment]
FWIW I just installed F33 Silverblue on a ThinkPad T495s and doing `rpm-ostree install fprintd fprintd-pam` makes the fingerprint sensor work.

**Comment by Kekun** [comment]
I'd love to see that be fixed, especially as `rpm-ostree install fprintd fprintd-pam` didn't work around the issue for me:

![Capture d’écran de 2021-01-02 21-52-10](https://user-images.githubusercontent.com/2536339/103466384-fea0d480-4d44-11eb-85fb-646a50378f47.png)

**Comment by YaLTeR** [comment]
> FWIW I just installed F33 Silverblue on a ThinkPad T495s and doing `rpm-ostree install fprintd fprintd-pam` makes the fingerprint sensor work.

To update this statement, the fingerprint works for logging in but for some reason doesn't work for `sudo` (as it does on Workstation). Would be great to have it working for both.

**Comment by tpopela** [comment]
With https://pagure.io/workstation-ostree-config/c/38252bc2a8ac7d1e2a753fc1bacd96701172ed1d?branch=master and https://pagure.io/workstation-ostree-config/c/00776d5bae6f8051ff359dec143451f70307336f?branch=master the fprintd-pam and fprintd should be part of the next Silverblue Rawhide compose.

**Comment by benzea** [comment]
> I'd love to see that be fixed, especially as `rpm-ostree install fprintd fprintd-pam` didn't work around the issue for me:

Hmm, that is an odd issue. The best guess I have is that fwupd was just updating the firmware for the device. If it is still happening, could you open a bug upstream (or ping me to get more information? Would help to add `Environment=G_MESSAGES_DEBUG=all` to fprintd.service using `systemctl edit`).

**Comment by benzea** [comment]
> To update this statement, the fingerprint works for logging in but for some reason doesn't work for `sudo` (as it does on Workstation). Would be great to have it working for both.

@YaLTeR, could you first check whether `with-fingerprint` is in `/etc/authselect/authselect.conf`. If it is not, try running `authselect --enable-feature with-fingerprint`, if it is, try just running `authselect` to regenerate the configuration.

**Comment by YaLTeR** [comment]
> > To update this statement, the fingerprint works for logging in but for some reason doesn't work for `sudo` (as it does on Workstation). Would be great to have it working for both.
> 
> @YaLTeR, could you first check whether `with-fingerprint` is in `/etc/authselect/authselect.conf`. If it is not, try running `authselect --enable-feature with-fingerprint`, if it is, try just running `authselect` to regenerate the configuration.

There's no such file actually, and `authselect` commands just complain that there's no existing configuration.

**Comment by pbrezina** [comment]
You can use this to enable authselect PAM stack: `authselect select sssd with-fingerprint --force`

**Comment by YaLTeR** [comment]
> You can use this to enable authselect PAM stack: `authselect select sssd with-fingerprint --force`

I did this and rebooted. Now `sudo` prompts for my fingerprint. However, failing a match once makes it not accept any more fingerprints until a timeout, despite what it says:

```
┌ ~
└─ sudo echo hi
Place your finger on the fingerprint reader
Failed to match fingerprint
Place your finger on the fingerprint reader
Place your finger on the reader again
^C^\Verification timed out
[sudo] password for yalter: 
┌ ~ 1
└─ sudo echo hi
Place your finger on the fingerprint reader
Failed to match fingerprint
Place your finger on the fingerprint reader
Place your finger on the reader again
Verification timed out
[sudo] password for yalter: 
┌ ~ 1
└─ fprintd-verify
Using device /net/reactivated/Fprint/Device/0
Listing enrolled fingers:
 - #0: right-index-finger
Verify started!
Verifying: right-index-finger
Verify result: verify-match (done)
┌ ~
└─ sudo echo hi
Place your finger on the fingerprint reader
hi
```

**Comment by benzea** [comment]
> However, failing a match once makes it not accept any more fingerprints until a timeout 

That is a known bug and a fix exists upstream. Also, entirely unrelated to the issue at hand here.

**Comment by YaLTeR** [comment]
Ah, okay, great.

Will this authselect configuration be enabled by default in F34 Silverblue considering fprintd should be included?

**Comment by DonGatoMalo** [comment]
fingerprint was working just fine , after install fprint-pam , 
but , now can't upgrade 
rpm-ostree upgrade: 
error: Sanity-checking final rpmdb: Didn't find package 'fprintd-1.90.9-1.fc33.x86_64'

**Comment by Honza0297** [comment]
That's very likely caused by https://discussion.fedoraproject.org/t/cannot-upgrade-silverblue/26652. Libsolv 0.7.17 and rpm-ostree do not cooperate well. You can find a solution there as well.

**Comment by DonGatoMalo** [comment]
> That's very likely caused by https://discussion.fedoraproject.org/t/cannot-upgrade-silverblue/26652. Libsolv 0.7.17 and rpm-ostree do not cooperate well. You can find a solution there as well.

Sorted . thx a lot

**Comment by tpopela** [comment]
Ok, I'm closing this issue as it has been resolved for Silverblue 34.
